app/api/services/informationextraction/getFiles.ts

```diff
@@ -71,7 +75,17 @@ async function getFilesForTraining(templates: ObjectIdSchema[], property: string
     { limit: MAX_TRAINING_FILES_NUMBER }
   )) as (FileType & FileEnforcedNotUndefined)[];
 
-  return getFilesWithAggregations(files);
+  const filesWithEntityValue = files.map(file => {
+    const entity = entities.find(e => e.sharedId === file.entity);
+    if (!entity?.metadata || !entity?.metadata[property]?.length) {
+      return file;
+    }
+    const [{ value }] = entity.metadata[property] || [{}];
+
+    return { ...file, label_text: value };
```

Comment:
Due to the type mismatch between the entity values and what the service expect, we should probably transform this for dates into the actual string the service expects. If we do it here, immediately when getting it from the entity, we can leave label_text as string after all.